src/main/asciidoc/reference/vault-secret-engines.adoc

```diff
@@ -0,0 +1,252 @@
+* Key reconfiguration
+* Encryption/Decryption/Rewrapping
+* HMAC computation
+* Signing and signature verification
+
+All operations within `transit` are centered around keys.
+The Transit engine supports the versioning of keys allowing you to pick a key type https://www.vaultproject.io/docs/secrets/transit[from a variety of key types].
+Note that depending on the key type may impose a limitation on which operations can used.
+
+Let's take a look at encryption and decryption.
+The following examples shows how to create a key and how to encrypt and decrypt data:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/Transit.java[tags=encryptSimple]
+----
+<1> First, we need a key to begin with.
+Each key requires the type to be specified. `aes128-gcm96` supports encryption, decryption, key derivation, and convergent encryption of which we need encryption and decryption for this example.
+<2> Next, we encrypt a `String` containing the plaintext that should be encrypted.
+The input `String` uses the default `Charset` to encode the string into its binary representation.
+Requesting a token returns a `VaultToken` which is used as value object for Vault tokens.
+The `encrypt` method returns Base64-encoded ciphertext, typically starting with `vault:`.
+<3> To decrypt ciphertext into plaintext, call the `decrypt` method.
+It decrypts the ciphertext and returns a `String` that is decoded using the default charset.
+====
+
+The preceeding example uses simple strings for cryptographic operations.
+While it's a simple approach, it bears the risk of charset misconfiguration and it is not binary-safe.
+Binary safety is required when the plaintext uses a binary representation for data such as images, compressed data or binary datastructures.
+
+To encrypt and decrypt binary data, use the `Plaintext` and `Ciphertext` value objects that can hold binary values:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/Transit.java[tags=encryptPlaintext]
+----
+<1> Assuming a key `my-aes-key` is already in place, we're encrypting the `Plaintext` object.
+In return, the `encrypt` method returns a `Ciphertext` object.
+<2> The `Ciphertext` object can be used directly for decryption and returns a `Plaintext` object.
+====
+
+`Plaintext` and `Ciphertext` come with a contextual object `VaultTransitContext`.
+It is used to supply a nonce value for https://www.vaultproject.io/docs/secrets/transit#convergent-encryption[convergent encryption] and for a context value to make use of key derivation.
+
+Transit allows for signing plaintext and verifying the signature for a given plaintext.
+Sign operations require an asymmetric key, typically using Elliptic Curve Cryptography or RSA.
+
+TIP: This tip explains how signatures work internally. +
+ +
+Signatures make use of the public/private key split to ensure authenticity. +
+The signer uses its private key to create a signature (as otherwise anybody would be able to sign messages in your name!).
```

Comment:
Change to "The signer uses its private key to create a signature. Otherwise, anybody would be able to sign messages in your name."